go/libkb/secret_store_file.go

```diff
@@ -98,8 +218,36 @@ func (s *SecretStoreFile) ClearSecret(username NormalizedUsername) error {
 	return nil
 }
 
+func (s *SecretStoreFile) clearSecretV2(username NormalizedUsername) error {
+	exists, err := FileExists(s.userpathV2(username))
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return nil
+	}
+
+	for i := 0; i < 3; i++ {
+		noise, err := RandBytes(1024 * 1024 * 2)
+		if err != nil {
+			return err
+		}
+		if err := ioutil.WriteFile(s.noisepathV2(username), noise, 0600); err != nil {
```

Comment:
I wonder if `ioutil.WriteFile` is doing the right thing here.  Is it writing the file in place, or opening a tempfile and swapping?

Reply:
Could you also do a little bit of research here to see if there are any best practices to employ? It could be some flags to `open` might be better here to ensure the writes are inplace as much as possible.

Reply:
It's a hard problem, because it's not just SSD, but the FS is an issue too:

https://askubuntu.com/questions/954848/using-shred-to-remove-files-on-ssd-lifespam

shred isn't applicable on ext4, for example...

Reply:
ioutil.WriteFile truncates if the file exists, but it is not doing any temp file swapping.

https://golang.org/src/io/ioutil/ioutil.go?s=2534:2602#L69

Reply:
https://en.wikipedia.org/wiki/Trim_(computing)